cobra/test/test_medium.py

```diff
@@ -143,6 +166,14 @@ class TestErrorsAndExceptions:
     def test_no_boundary_reactions(self, empty_model):
         assert medium.find_boundary_types(empty_model, 'e', None) == []
 
-    def test_no_boundary_reactions(self, empty_model):
+    def test_no_names_or_boundary_reactions(self, empty_model):
         with pytest.raises(RuntimeError):
             medium.find_external_compartment(empty_model)
+
+    def test_bad_exchange(self, model):
```

Comment:
I'd suggest to use parametrization:

```python
    @pytest.parametrize("metabolite", [
        pytest.param(Metabolite("baddy", compartment="nonsense"), mark=pytest.mark.raises(ValueError)),
        Metabolite("goody", compartment="e")
    ])
    def test_bad_exchange(self, model, metabolite): 
        rxn = model.add_boundary(metabolite)
        assert isinstance(rxn, Reaction)
```

Reply:
That does not work:

```
cobra/test/test_medium.py:164: in <module>
    class TestErrorsAndExceptions:
cobra/test/test_medium.py:175: in TestErrorsAndExceptions
    mark=pytest.mark.raises(ValueError)),
../../anaconda3/lib/python3.6/site-packages/_pytest/mark/structures.py:241: in __call__
    store_mark(func, self.mark)
../../anaconda3/lib/python3.6/site-packages/_pytest/mark/structures.py:276: in store_mark
    obj.pytestmark = get_unpacked_marks(obj) + [mark]
E   TypeError: can't set attributes of built-in/extension type 'ValueError'
```

Using `pytest.mark.xfail(raises=ValueError)` does not work either.

Reply:
My bad, need to make sure that `pytest-raises` is installed. Then it should be:
```python
pytest.param(Metabolite("baddy", compartment="nonsense"), mark=pytest.mark.raises(exception=ValueError))
```
with keyword argument.

Reply:
Okay, what about just refactoring that into two tests (one for good metabolites and one for bad) and just using the normal parameterize? This way we don't have to introduce new test dependencies.